Source/DafnyCore/AST/SystemModuleManager.cs

```diff
@@ -306,6 +305,48 @@ public UserDefinedType ArrayType(int dims, Type arg, bool allowCreationOfNewClas
   }
 
   // Initialize fp64 type and add to system module on first reference
+  public void EnsureFp32TypeInitialized(ProgramResolver programResolver) {
+    EnsureFp32TypeInitialized();
+
+    var fp32TypeDecl = valuetypeDecls[(int)ValuetypeVariety.Fp32];
+
+    // Add fp32 to system module if not already there
+    if (!SystemModule.SourceDecls.Contains(fp32TypeDecl)) {
+      SystemModule.SourceDecls.Add(fp32TypeDecl);
+    }
+
+    // Register fp32 in systemNameInfo.TopLevels for name resolution
+    systemNameInfo.TopLevels.TryAdd("fp32", fp32TypeDecl);
+
+    // Add fp32 members to classMembers dictionary
+    var memberDictionary = fp32TypeDecl.Members
+      .GroupBy(member => member.Name)
+      .ToDictionary(g => g.Key, g => g.First());
+    programResolver.AddSystemClass(fp32TypeDecl, memberDictionary);
+  }
+
+  private void EnsureFp32TypeInitialized() {
+    var fp32TypeDecl = valuetypeDecls[(int)ValuetypeVariety.Fp32];
+
+    // Check if already initialized by looking for "Min" member
+    if (fp32TypeDecl.Members.Any(m => m.Name == "Min")) {
+      return;
+    }
+
+    // Initialize both fp32 and fp64 members together (they share implementation)
```

Comment:
Since this code does something for both `fp32` and `fp64`, the name of the enclosing method, `EnsuresFp32TypeInitialized`, seems less appropriate. Can you rename it or perhaps move the fp32/fp64 member initialization to a different method?